asm-utils/src/reachability.rs

```diff
@@ -141,13 +141,20 @@ fn extract_replacements<R: Register, F: FunctionOpKind>(
         .iter()
         .filter_map(|s| match s {
             Statement::Directive(dir, args) if dir.as_str() == ".set" => {
-                if let [Argument::Expression(Expression::Symbol(from)), Argument::Expression(Expression::Symbol(to))] = &args[..]
-                {
-                    Some((from.to_string(), to.to_string()))
-                } else {
-                    panic!();
+                match &args[..] {
+                    [Argument::Expression(Expression::Symbol(from)), Argument::Expression(Expression::Symbol(to))] =>
+                    {
+                        Some((from.to_string(), to.to_string()))
+                    },
+                    [Argument::Expression(Expression::Symbol(_)), Argument::Expression(Expression::Number(_))] => {
```

Comment:
what exactly falls here and needs to be handled elsewhere?

Reply:
Clarified comment.

Reply:
But this function handle aliases: making one label alias for another. The variant of `.set` directive I am using now is not an alias, so it is not handled here, but can't fail here either, because it is valid.